# Superstore Profitability Analysis Dashboard

### Objectives:
* To identify where the **business is making money**, **where it is losing profit**, and what **actions can improve profitability**.
* To showcase all findings/insights into an **easy-to-understand dashboard**

---

### Install libraries:

In [2]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

### Load the dataset:

In [3]:
df = pd.read_csv('/Users/saadult/superstore-profitability/data/SampleSuperstore.csv')

In [4]:
df.shape

(9994, 13)

The dataset has 9,994 rows and 13 columns.

---

### Initial dataset inspections:

In [5]:
df.head()

,Ship Mode,Segment,Country,City,State,Postal Code,Region,Category,Sub-Category,Sales,Quantity,Discount,Profit
0,Second Class,Consumer,United States,Henderson,Kentucky,42420,South,Furniture,Bookcases,261.9600,2,0.00,41.9136
1,Second Class,Consumer,United States,Henderson,Kentucky,42420,South,Furniture,Chairs,731.9400,3,0.00,219.5820
2,Second Class,Corporate,United States,Los Angeles,California,90036,West,Office Supplies,Labels,14.6200,2,0.00,6.8714
3,Standard Class,Consumer,United States,Fort Lauderdale,Florida,33311,South,Furniture,Tables,957.5775,5,0.45,-383.0310
4,Standard Class,Consumer,United States,Fort Lauderdale,Florida,33311,South,Office Supplies,Storage,22.3680,2,0.20,2.5164


In [6]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9994 entries, 0 to 9993
Data columns (total 13 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   Ship Mode     9994 non-null   object 
 1   Segment       9994 non-null   object 
 2   Country       9994 non-null   object 
 3   City          9994 non-null   object 
 4   State         9994 non-null   object 
 5   Postal Code   9994 non-null   int64  
 6   Region        9994 non-null   object 
 7   Category      9994 non-null   object 
 8   Sub-Category  9994 non-null   object 
 9   Sales         9994 non-null   float64
 10  Quantity      9994 non-null   int64  
 11  Discount      9994 non-null   float64
 12  Profit        9994 non-null   float64
dtypes: float64(3), int64(2), object(8)
memory usage: 1015.1+ KB


The dataset has both categorical and numerical values.

In [8]:
df.describe().round(2)

,Postal Code,Sales,Quantity,Discount,Profit
count,9994.00,9994.00,9994.00,9994.00,9994.00
mean,55190.38,229.86,3.79,0.16,28.66
std,32063.69,623.25,2.23,0.21,234.26
min,1040.00,0.44,1.00,0.00,-6599.98
25%,23223.00,17.28,2.00,0.00,1.73
50%,56430.50,54.49,3.00,0.20,8.67
75%,90008.00,209.94,5.00,0.20,29.36
max,99301.00,22638.48,14.00,0.80,8399.98


<u>Insights:</u>
* Sales and profit distributions are highly skewed. Average sales are around $230, while the maximum sale exceeds $22,000. This indicates the presence of large outliers transactions.
* Average profit is $28.66, but the minimum profit reaches -$6,599.98. This suggests some products, discounts or regions maybe generating significant losses that require deeper investigation.

---

### Data validation:

In [26]:
# Check missing values
df.isnull().sum()

Ship Mode         0
Segment           0
Country           0
City              0
State             0
Postal Code       0
Region            0
Category          0
Sub-Category      0
Sales             0
Quantity          0
Discount          0
Profit            0
Profit Margin     0
Profit Status     0
Discount Level    0
dtype: int64

In [27]:
# Check duplicate rows/values
df.duplicated().sum()

np.int64(17)

<u>Insights:</u>
* The dataset has no missing values across all key business and sales-related fields.
* There are 17 duplicate records identified out of nearly 10,000 rows.

In [14]:
# Show duplicate rows
df[df.duplicated(keep=False)].sort_values(by=list(df.columns))

,Ship Mode,Segment,Country,City,State,Postal Code,Region,Category,Sub-Category,Sales,Quantity,Discount,Profit
6409,First Class,Consumer,United States,Houston,Texas,77041,Central,Office Supplies,Paper,47.952,3,0.2,16.1838
8032,First Class,Consumer,United States,Houston,Texas,77041,Central,Office Supplies,Paper,47.952,3,0.2,16.1838
5493,Same Day,Home Office,United States,San Francisco,California,94122,West,Office Supplies,Labels,41.400,4,0.0,19.8720
5905,Same Day,Home Office,United States,San Francisco,California,94122,West,Office Supplies,Labels,41.400,4,0.0,19.8720
2807,Second Class,Consumer,United States,Seattle,Washington,98115,West,Office Supplies,Paper,12.960,2,0.0,6.2208
8095,Second Class,Consumer,United States,Seattle,Washington,98115,West,Office Supplies,Paper,12.960,2,0.0,6.2208
8457,Second Class,Corporate,United States,Chicago,Illinois,60653,Central,Office Supplies,Binders,3.564,3,0.8,-6.2370
9477,Second Class,Corporate,United States,Chicago,Illinois,60653,Central,Office Supplies,Binders,3.564,3,0.8,-6.2370
8533,Standard Class,Consumer,United States,Detroit,Michigan,48227,Central,Furniture,Chairs,389.970,3,0.0,35.0973
9262,Standard Class,Consumer,United States,Detroit,Michigan,48227,Central,Furniture,Chairs,389.970,3,0.0,35.0973


<u>Insight:</u>
17 fully identical rows are found in the dataset, which is only around 0.17% of the total data. Since the dataset doesn’t include a unique transaction or order ID, I decided to **keep them** because they could still represent **legitimate repeated purchases**. Also, the number of duplicates is too small to significantly affect the overall analysis and insights.

---

### Basic Business Metrics:
Calculating high-level business KPIs to understand the company’s overall sales, profitability, and transaction volume.

<u>Questions Being Answered:</u>
- How much total sales did the company generate?
- How much total profit was earned?
- What is the overall profit margin?
- How many products were sold?

In [18]:
total_sales = df["Sales"].sum().round(2)
total_profit = df["Profit"].sum().round(2)
profit_margin = total_profit / total_sales
total_quantity = df["Quantity"].sum().round(2)

print(f"Total sales: {total_sales}")
print(f"Total profit: {total_profit}")
print(f"Profit margin: {profit_margin}")
print(f"Total quantity: {total_quantity}")

Total sales: 2297200.86
Total profit: 286397.02
Profit margin: 0.1246721716794064
Total quantity: 37873


Despite generating over $2.29M in sales, the company achieved only a 12.47% profit margin, suggesting that strong revenue does not always translate into high profitability.

---

### Profit by Region:
Analyzing profitability across geographic regions to identify the strongest and weakest performing areas.

<u>Questions Being Answered:</u>
- Which regions generate the highest profit?
- Which regions underperform financially?
- Are there geographic differences in profitability?

In [21]:
df.groupby("Region")["Profit"].sum().round(2).sort_values(ascending=False)

Region
West       108418.45
East        91522.78
South       46749.43
Central     39706.36
Name: Profit, dtype: float64

The West region generated the highest profit, while the Central region significantly underperformed, highlighting notable geographic differences in profitability across the business.

---

### Profit by Category:
Evaluating profitability across major product categories to determine which categories contribute most to business success.

<u>Questions Being Answered:</u>
- Which product categories are most profitable?
- Which categories generate weaker returns?
- Where should the business focus growth efforts?


In [22]:
df.groupby("Category")["Profit"].sum().round(2).sort_values(ascending=False)

Category
Technology         145454.95
Office Supplies    122490.80
Furniture           18451.27
Name: Profit, dtype: float64

Technology and Office Supplies were the strongest profit-driving categories, while Furniture generated significantly lower returns despite likely contributing substantial sales volume.

---

### Profit by Sub-category:
Drilling deeper into product performance to identify specific sub-categories driving profit or causing losses.

<u>Questions Being Answered:</u>
- Which sub-categories are the most profitable?
- Which sub-categories are losing money?
- What specific product groups require business attention?


In [23]:
df.groupby("Sub-Category")["Profit"].sum().round(2).sort_values()

Sub-Category
Tables        -17725.48
Bookcases      -3472.56
Supplies       -1189.10
Fasteners        949.52
Machines        3384.76
Labels          5546.25
Art             6527.79
Envelopes       6964.18
Furnishings    13059.14
Appliances     18138.01
Storage        21278.83
Chairs         26590.17
Binders        30221.76
Paper          34053.57
Accessories    41936.64
Phones         44515.73
Copiers        55617.82
Name: Profit, dtype: float64

Tables, Bookcases, and Supplies were the only sub-categories generating losses, while Copiers and Phones emerged as the strongest profit contributors, revealing clear opportunities for product-level optimization.

---

### Discount and Profit Relationship:
Analyzing the relationship between discounts and profitability to determine whether aggressive discounting impacts business performance.

<u>Questions Being Answered:</u>
- Do higher discounts reduce profit?
- Is there a relationship between discounting and financial losses?
- Should the business review its discount strategy?


In [24]:
df[["Discount", "Profit"]].corr()

,Discount,Profit
Discount,1.000000,-0.219487
Profit,-0.219487,1.000000


There is a noticeable negative relationship between discounts and profit, suggesting that higher discount levels may contribute to reduced profitability and potential financial losses.

---

### Create New Fields:
Creating calculated fields and business-friendly categories that will improve analysis and dashboard interactivity in Tableau.

<u>Questions Being Answered:</u>
- How can we simplify the analysis for business users?
- Which calculated metrics can improve reporting?
- How can we better categorize discount and profit behavior?

In [25]:
df["Postal Code"] = df["Postal Code"].astype(str)
df["Profit Margin"] = df["Profit"] / df["Sales"]

df["Profit Status"] = df["Profit"].apply(
    lambda x: "Profit" if x >= 0 else "Loss"
)

def discount_level(discount):
    if discount == 0:
        return "No Discount"
    elif discount <= 0.2:
        return "Low Discount"
    elif discount <= 0.4:
        return "Medium Discount"
    else:
        return "High Discount"

df["Discount Level"] = df["Discount"].apply(discount_level)

df.head()

,Ship Mode,Segment,Country,City,State,Postal Code,Region,Category,Sub-Category,Sales,Quantity,Discount,Profit,Profit Margin,Profit Status,Discount Level
0,Second Class,Consumer,United States,Henderson,Kentucky,42420,South,Furniture,Bookcases,261.9600,2,0.00,41.9136,0.1600,Profit,No Discount
1,Second Class,Consumer,United States,Henderson,Kentucky,42420,South,Furniture,Chairs,731.9400,3,0.00,219.5820,0.3000,Profit,No Discount
2,Second Class,Corporate,United States,Los Angeles,California,90036,West,Office Supplies,Labels,14.6200,2,0.00,6.8714,0.4700,Profit,No Discount
3,Standard Class,Consumer,United States,Fort Lauderdale,Florida,33311,South,Furniture,Tables,957.5775,5,0.45,-383.0310,-0.4000,Loss,High Discount
4,Standard Class,Consumer,United States,Fort Lauderdale,Florida,33311,South,Office Supplies,Storage,22.3680,2,0.20,2.5164,0.1125,Profit,Low Discount


Creating calculated fields such as **Profit Margin**, **Profit Status**, and **Discount Level** improves the clarity of analysis, simplifies business reporting, and enables more interactive and meaningful Tableau visualizations.

---

### Export Clean Dataset for Tableau:
Exporting the cleaned and transformed dataset for use in Tableau dashboard development.

In [28]:
df.to_csv("../data/cleaned_superstore.csv", index=False)